Makefile.system

```diff
@@ -497,6 +497,14 @@ endif
 endif
 endif
 
+ifdef DYNAMIC_LIST
+override DYNAMIC_CORE = PRESCOTT $(DYNAMIC_LIST)
+XCCOMMON_OPT = -DDYNAMIC_LIST -DDYN_PRESCOTT
```

Comment:
This uses PRESCOTT as the fallback for everything. 

I think there are two things to reason about here.

1. Should we expect the user to know what the lowest common denominator is when using the DYNAMIC_LIST option, instead of specifying it for them? I think the answer is yes. 

Then we have the error handling questin, , what to do in the case a CPU is detected at runtime which is worse in its capabilities than what the user specified as the minimum CPU. 

One thing to do would be what you are doing here, which is the safe thing, have everything fallback to PRESCOTT, the other would be to actually crash with an error which says that the library does not have support for the architecture detected.

I would argue that we should do the latter, since we should not paper over the user's poorly chosen list of arches. You should only use DYNAMIC_LIST if you know what you are doing.

2. Given the list of dynamic architectures given by the user, should we make an attempt to match each architecture that is NOT present in the list to fall back to its "nearest" archiecture in the DYNAMIC_LIST or should they all fallback to the lowest common denominator?

Doing this correctly would require specifying for each architecture a descending chain and building the logic to intersect this chain with DYNAMIC_LIST and picking the top element.

Reply:
Fair point about crashing and the user maybe far removed from the decision maker.  I was taking a narrower view of this option, as an expert only thing.  

Would you be open to making the default fallback also configurable instead of hardcoding it to be PRESCOTT?

Reply:
I could add a DYNAMIC_FALLBACK option, but I am not really convinced yet that it would be useful. After all this would be needed only for cpus not on the list, and if we want to avoid crashing it would have to be suitable for any old system that OpenBLAS supports - which basically leaves only Prescott I think. Setting up optimum fallback for each cpu would mean long strings of ifndef(THISCPU) ... ifdef(PREVIOUS_GENERATION)...elseif(NEXT_OLDER)....elseif...elseif...

Reply:
Fair enough. what other kind of feedback are you looking for before refining/finalizing this PR?

Reply:
Well, do you think it would even address your wish for a short, user-defined list (the fallback issue nonwithstanding, perhaps I could try to fallback to one or two intermediate types, e.g. Sandybridge and Nehalem before defaulting to Prescott) ?

Reply:
Yes I think it will. 
Your argument for PRESCOTT as the fallback is cogent and I agree with it. I was trying to over-optimize. 
Let me try this patch on my internal build and get back to you later today.